### PR TITLE
stargate: Add `proto-gen` build step

### PR DESCRIPTION
**Pull request by youngjoon-lee**
Cosmos SDK uses [a modified version](https://github.com/regen-network/cosmos-proto) of [gogoprotobuf](https://github.com/gogo/protobuf). However, Cosmos doesn't provide any guide how to run `protoc`. They just say we have to refer to their Makefile: https://docs.cosmos.network/master/migrations/app_and_modules.html.

So,
I've written our own Makefile by referring to scripts in [cosmos-sdk v0.42.4](https://github.com/cosmos/cosmos-sdk/tree/v0.42.4) (`Makefile` and `scripts/`).

For usages, please see changes in the `.github/workflows/ci.yml`.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -38,6 +39,16 @@ lint:
...
+########################################
+### Protobuf
+
+proto-gen: proto-update-deps
+	@echo "Generating Protobuf files"
+	$(DOCKER) run --rm -v $(CURDIR):/workspace --workdir /workspace tendermintdev/sdk-proto-gen sh ./scripts/protocgen.sh
+
+proto-update-deps:
+	GO111MODULE=off go get github.com/stormcat24/protodep
+	protodep up
```
Comment (youngjoon-lee) [inline Makefile:51]: For managing 3rd-party proto files (depedencies), I'm trying to use [protodep](https://github.com/stormcat24/protodep). It's my first time to try it, but it's simple to use. Please see details in the `protodep.toml` file below.

File: Makefile [hunk 2]
```diff
@@ -2,6 +2,7 @@ VERSION := $(shell echo $(shell git describe --tags) | sed 's/^v//')
 COMMIT := $(shell git log -1 --format='%H')
 GOBIN ?= $(GOPATH)/bin
 GOSUM := $(shell which gosum)
+DOCKER := $(shell which docker)
```
Comment (gyuguen) [inline Makefile:5]: To install Docker!

**Review by gyuguen — APPROVED** [review]
👍

File: Makefile (hunk 2 above)
Comment (youngjoon-lee) [reply Makefile:5]: Yeah. The `tendermintdev/sdk-proto-gen` image is used, but I coudn't find its Dockerfile. When I inspect the image using [dive](https://github.com/wagoodman/dive), I could find that the image contains the following layers:
image
But, it's not clear yet. Let's use this docker strategy for now, and consider alternative ways after making Stargate work.

**Comment by youngjoon-lee** [comment]
@gyuguen Requesting your review again. Now, CI works.
7b95ef3a76cbb48ef1f9191f08ff3b25bcd8b20e
I started ssh-agent and register my SSH private key, because [protodep](https://github.com/stormcat24/protodep) requires SSH for fetching codes from github.

**Review by gyuguen — APPROVED** [review]
👍

**Pull request merged**